lighthouse-core/gather/gatherers/dobetterweb/response-compression.js

```diff
@@ -58,7 +58,11 @@ class ResponseCompression extends Gatherer {
 
     const driver = options.driver;
     return Promise.all(textRecords.map(record => {
-      return driver.getRequestContent(record.requestId).then(content => {
+      const contentPromise = driver.getRequestContent(record.requestId);
+      const timeoutPromise = new Promise((resolve,reject)=>{
```

Comment:
we should util a promise race timeout like this and do it everywhere

this brings up a good point, should we race all gatherer afterPasses at some max timeout?